### PR TITLE
Moves the validation of client dni to the service layer

### PR DESCRIPTION
**Pull request by PatoRocaspana**

### CODE REVIEW

File: RentACarWebAPI/Services/ClientService.cs [hunk 1]
```diff
@@ -1,13 +1,49 @@
...
+        }
+
+        public override Client Update(Client newClient, int id)
+        {
+            var existingClient = _clientRepository.Get(id);
+
+            if (existingClient is null)
+                return null;
+
+            var clientList = _clientRepository.GetAll();
```
Comment (PatoRocaspana) [inline RentACarWebAPI/Services/ClientService.cs:36]: In a real case, maybe with a bigger db of clients, Is it good to use the `Get(id)` method and then use `GetAll()` method?
Or should i `GetAll()` clients and then iterate there?

Comment (jrocaspana) [reply RentACarWebAPI/Services/ClientService.cs:36]: you never want to load an entire table in memory just to perform a check (there may be some scenario but it's an edge one). whatever you plan to check in memory you should be able to do it at the database (making several queries is always an option)

Comment (PatoRocaspana) [reply RentACarWebAPI/Services/ClientService.cs:36]: so, in this case, when i have the db i should to move this, right?
when i say **this** i mean the validation about if a client already have same _dni_ ..
Or maybe add another endpoint to get a client by _dni_ ?

Comment (jrocaspana) [reply RentACarWebAPI/Services/ClientService.cs:36]: correct, you would query the database to see if a record exists with the same DNI and act accordingly

**Pull request merged**